File: IDSFintechPortal.Api/Models/Repository.cs
namespace IDSFintechPortal.Api.Models
{
    public class Repository
    {
        public int Id { get; set; }
        public int ProductId { get; set; }
        public string RepoName { get; set; } = string.Empty;
        public string GithubUrl { get; set; } = string.Empty;
        public string? MainBranch { get; set; }
        public string? Description { get; set; }
    }
}

File: ids-fintech-portal-frontend/src/types/Repository.ts
export interface Repository  {
    id: number;
    productId: number;
    repoName: string;
    githubUrl: string;
    mainBranch: string | null;
    description: string | null;
}

export interface CreateRepositoryDto {
    productId: number;
    repoName: string;
    githubUrl: string;
    mainBranch?: string;
    description?: string;
}